<launch>
    <arg name="dsd_file" default="main.dsd" />
    <arg name="tf_prefix" default=""/>
    <arg name="sim" default="false" />

    <let if="$(env IS_ROBOT false)" name="taskset" value="taskset -c 2"/>
    <let unless="$(env IS_ROBOT false)" name="taskset" value=""/>

    <node pkg="bitbots_body_behavior" exec="body_behavior" output="screen" launch-prefix="$(var taskset)">
        <param name="dsd_file" value="$(var dsd_file)"/>
        <param name="actionlib_client_sub_queue_size" value="-1"/>
        <param name="actionlib_client_sub_queue_size" value="-1"/>
        <param name="actionlib_server_sub_queue_size" value="-1"/>
        <param from="$(find-pkg-share bitbots_body_behavior)/config/body_behavior.yaml"/>
        <param from="$(find-pkg-share bitbots_body_behavior)/config/animations.yaml"/>
        <param name="odom_frame" value="$(var tf_prefix)odom"/>
        <param name="map_frame" value="$(var tf_prefix)map"/>
        <param name="ball_frame" value="$(var tf_prefix)ball"/>
        <param name="base_footprint_frame" value="$(var tf_prefix)base_footprint"/>
        <param name="use_sim_time" value="$(var sim)" />
    </node>
</launch>
